File: Luogu/Greedy/P1090.cc
/* P1090.cc */

#include <iostream>
#include <queue>
#include <vector>

struct cmp {
    bool operator()(const int& a, const int& b) {
        return a > b;
    }
};

int main(void) {
    std::priority_queue<int, std::vector<int>, cmp> pq;        
    int n = 0, x = 0;
    int sum = 0;
    std::cin >> n;

    for (int i = 0; i < n; i ++) {
        std::cin >> x;
        pq.push(x);
    }

    while (pq.size() > 1) {
        int bunch_1 = pq.top();
        pq.pop();
        int bunch_2 = pq.top();
        pq.pop();
        sum += bunch_1 + bunch_2;
        pq.push(bunch_1 + bunch_2);
    }

    std::cout << sum << std::endl;

    return 0;
}

File: Luogu/Greedy/P1090.py
from queue import PriorityQueue

n = int(input())
in_str = input()
sum = 0
pq = PriorityQueue()
bunches_str = list(in_str.split(" "))
bunches = []

for i in range(n):
    if ('\0' == bunches_str[i]): break
    bunches.append(int(bunches_str[i]))

for bunch in bunches:
    pq.put(bunch)

while (pq.qsize() > 1):
    bunch_1 = pq.get()
    bunch_2 = pq.get()
    sum += (bunch_1 + bunch_2)
    pq.put(bunch_1 + bunch_2)

print(sum)
